src/components/About/Blog/BlogPost.js

```diff
@@ -0,0 +1,339 @@
+import React from 'react';
+import _Card from '@material-ui/core/Card';
+import CardMedia from '@material-ui/core/CardMedia';
+import CardContent from '@material-ui/core/CardContent';
+import _Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import {
+  FacebookShareButton,
+  TwitterShareButton,
+  FacebookIcon,
+  TwitterIcon,
+} from 'react-share';
+import styled, { css } from 'styled-components';
+import Fade from '@material-ui/core/Fade';
+import './Blog.css';
+import dateFormat from 'dateformat';
+import htmlToText from 'html-to-text';
+import renderHTML from 'react-render-html';
+import susi from '../../../images/susi-logo.svg';
+import PropTypes from 'prop-types';
+
+const allCategories = [
+  'FOSSASIA',
+  'GSoC',
+  'SUSI.AI',
+  'Tutorial',
+  'Android',
+  'API',
+  'App generator',
+  'CodeHeat',
+  'Community',
+  'Event',
+  'Event Management',
+  'loklak',
+  'Meilix',
+  'Open Event',
+  'Phimpme',
+  'Pocket Science Lab',
+  'yaydoc',
+];
+
+const arrDiff = (a1, a2) => {
+  let a = [],
+    diff = [];
+  for (let f = 0; f < a1.length; f++) {
+    a[a1[f]] = true;
+  }
+  for (let z = 0; z < a2.length; z++) {
+    if (a[a2[z]]) {
+      delete a[a2[z]];
+    } else {
+      a[a2[z]] = true;
+    }
+  }
+  for (let k in a) {
+    diff.push(k);
+  }
+  return diff;
+};
+
+const Overlay = styled.div`
+  position: relative;
+  left: 0rem;
+  background: rgba(0, 0, 0, 0.54);
+  width: 100%;
+  padding: 1rem;
+  margin-top: -3.5rem;
+`;
+
+const LinkStyle = css`
+  font-size: 0.875rem;
+  text-decoration: none;
+  color: rgba(255, 255, 255, 0.54);
+`;
+
+const FlexBox = styled.div`
+  display: flex;
+  align-items: baseline;
+  color: rgba(51, 51, 51, 0.7);
+  line-height: 1.563rem;
+  margin-right: 2rem;
+`;
+const Card = styled(_Card)`
+  position: relative;
+`;
+
+const CustomTypography = styled(_Typography)`
+  margin-bottom: 2rem;
+  color: rgba(0, 0, 0, 0.54);
+`;
+
+const Typography = styled(_Typography)`
+  font-family: 'Raleway';
+`;
+
+const OverlayLink = styled.a`
+  font-family: 'Raleway';
+  &&& {
+    ${LinkStyle};
+  }
+`;
+
+const BlogPostContainer = styled.div`
+  padding: 1rem;
+`;
+
+const Icon = styled.i`
+  margin-left: 0.5rem;
+  padding-right: 0.625rem;
+`;
+
+const BlogFooter = styled.div`
+  padding: 3rem;
+  background: #f7f7f7;
+  display: flex;
+  flex-wrap: wrap;
+`;
+
+const BlogFooterLink = styled.a`
+  white-space: nowrap;
+  cursor: pointer;
+  text-decoration: none;
+  color: rgba(0, 0, 0, 0.87);
+  height: 1.875rem;
+  &:hover {
+    text-decoration: underline;
+    color: rgba(51, 51, 51, 0.7);
+  }
+`;
+
+const SocialButtons = styled.div`
+  width: 100%;
+  display: flex;
+  padding: 0.625rem 0 1.25rem 0.625rem;
+`;
+
+class BlogPost extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      expandedElements: [],
```

Comment:
Why is this required as array? Bool variable named `isExpanded` should be fine IMO.